From c0ef420cedbf102f8017e0f7302c09a487b18d42 Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@gmail.com>
Date: Wed, 23 Mar 2011 13:42:18 -0400
Subject: [PATCH 2/2] gmapping_export:  fix compile

- Fix order of compile arguments.
- Append libs as LIBS not LDFLAGS to ensure correct order.
- Set LD_LIBRARY_PATH when compiling so NEEDED so's can be found.
---
 build_tools/Makefile.generic-shared-object |    4 ++--
 gridfastslam/Makefile                      |    2 +-
 log/Makefile                               |    2 +-
 scanmatcher/Makefile                       |    2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/build_tools/Makefile.generic-shared-object b/build_tools/Makefile.generic-shared-object
index 12b1457..5872410 100644
--- a/build_tools/Makefile.generic-shared-object
+++ b/build_tools/Makefile.generic-shared-object
@@ -35,14 +35,14 @@ ifeq ($(MACOSX),1)
 	@$(PRETTY) "$(CXX) $(LDFLAGS)  -dynamiclib $(OBJS) $(COBJS) -L$(LIBDIR) $(LIBS) -install_name $@ -o $@"
 endif
 ifeq ($(LINUX),1)
-	@$(PRETTY) "$(CXX) $(LDFLAGS) -fPIC -shared $(OBJS) $(COBJS) -L $(LIBDIR) $(LIBS) -o $@"
+	@$(PRETTY) "LD_LIBRARY_PATH=$(LIBDIR) $(CXX) -fPIC -shared $(OBJS) $(COBJS) -L $(LIBDIR) $(LIBS) -o $@" $(LDFLAGS) 
 	@if ! $(PRETTY) "$(TESTLIB) $(SONAME)"; then $(MESSAGE) "Testing of $(SONAME) failed."; rm $(SONAME); exit 1; fi;
 endif
 
 # Generazione delle applicazioni
 $(BINDIR)/%: %.o $(SONAME)
 	@$(MESSAGE) "Linking application `basename "$@"`"
-	@$(PRETTY) "$(CXX) $(LDFLAGS) -L$(LIBDIR) $(LIBS) -l$(LIBNAME) $< -o $@"
+	@$(PRETTY) "LD_LIBRARY_PATH=$(LIBDIR) $(CXX) -L$(LIBDIR) $(LIBS) -l$(LIBNAME) $< -o $@" $(LDFLAGS) 
 
 #Generazione dei moc files
 moc_%.cpp:              %.h
diff --git a/gridfastslam/Makefile b/gridfastslam/Makefile
index 87ec52a..9b77f26 100644
--- a/gridfastslam/Makefile
+++ b/gridfastslam/Makefile
@@ -2,7 +2,7 @@ OBJS= gridslamprocessor_tree.o motionmodel.o gridslamprocessor.o gfsreader.o
 APPS= gfs2log gfs2rec gfs2neff #gfs2stat
 
 #LDFLAGS+= -lutils -lsensor_range -llog -lscanmatcher -lsensor_base -lsensor_odometry $(GSL_LIB)
-LDFLAGS+= -lutils -lsensor_range -llog -lscanmatcher -lsensor_base -lsensor_odometry
+LIBS += -lutils -lsensor_range -llog -lscanmatcher -lsensor_base -lsensor_odometry
 #CPPFLAGS+=-I../sensor $(GSL_INCLUDE)
 CPPFLAGS+=-I../sensor
 
diff --git a/log/Makefile b/log/Makefile
index 5ba38e3..63d5d56 100644
--- a/log/Makefile
+++ b/log/Makefile
@@ -1,7 +1,7 @@
 OBJS= configuration.o carmenconfiguration.o sensorlog.o sensorstream.o
 APPS= log_test log_plot scanstudio2carmen rdk2carmen
 
-LDFLAGS+= -lsensor_base -l sensor_odometry -l sensor_range
+LIBS += -lsensor_base -l sensor_odometry -l sensor_range
 CPPFLAGS+= -I../sensor 
 
 -include ../global.mk
diff --git a/scanmatcher/Makefile b/scanmatcher/Makefile
index a9aa1d1..bf2ff6e 100644
--- a/scanmatcher/Makefile
+++ b/scanmatcher/Makefile
@@ -2,7 +2,7 @@ OBJS= smmap.o scanmatcher.o scanmatcherprocessor.o eig3.o
 APPS= scanmatch_test icptest
 
 #LDFLAGS+= $(GSL_LIB) -lutils -lsensor_range -llog
-LDFLAGS+= -lutils -lsensor_range -lsensor_base -llog
+LIBS += -lutils -lsensor_range -lsensor_base -llog
 #CPPFLAGS+=-I../sensor $(GSL_INCLUDE)
 CPPFLAGS+=-I../sensor
 
-- 
1.7.3.4

